MaterialLib/SolidModels/MFront/MFront.cpp

```diff
@@ -293,19 +291,44 @@ MFront<DisplacementDim>::integrateStress(
 
     auto v = mgis::behaviour::make_view(behaviour_data);
 
-    auto const eps_prev_MFront = OGSToMFront(eps_prev);
+    // rotation tensor
+    auto const Q = [this, &x]() -> KelvinMatrixType<DisplacementDim> {
+        if (!_local_coordinate_system)
+        {
+            return KelvinMatrixType<DisplacementDim>::Identity();
+        }
+        return fourthOrderRotationMatrix(
+            _local_coordinate_system->transformation<DisplacementDim>(x));
+    }();
+
+    auto const eps_prev_MFront =
+        OGSToMFront(Q.transpose()
+                        .template topLeftCorner<
+                            KelvinVectorDimensions<DisplacementDim>::value,
+                            KelvinVectorDimensions<DisplacementDim>::value>() *
```

Comment:
is the topLeftCorner stuff needed now that we have a 2D (4x4) variant of Q?

Reply:
To be brief, yes.